code-challenges.test.js

```diff
@@ -14,39 +14,167 @@
+    })
+
+})
+
+
 
-const letterA = "a"
-// Expected output: ["Mango", "Apricot", "Peach"]
-const letterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
 
 // b) Create the function that makes the test pass.
+// Psuedo code
+// declare a function
+// use the words array and the letter as your paramaters
+// iterate through the array checking for the string that matches your string
+// if it matches return that word in youre new array 
+// declare a new function thatll be filtered words
+// use .filter method to make a new array 
+// use .tolowercase to include capital letters
+// use .includes to check if the word has the specified letter
+// return filteredWords
+
+const wordsWithLetter = (wordsArray, letter) => {
+    const filteredWords = wordsArray.filter(word => word.toLowerCase().includes(letter.toLowerCase()))
+    return filteredWords 
+}
+
+// Test Suites: 1 passed, 1 total
+// Tests:       5 passed, 5 total
+// Snapshots:   0 total
+// Time:        0.291 s, estimated 1 s
+// Ran all test suites.
+
+
 
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-const hand4 = [7, 2, 7, 2, 7]
-// Expected output: true
+describe("fullHouse", () => {
+    const hand1 = [5, 5, 5, 3, 3]
+    // Expected output: true
+    const hand2 = [5, 5, 3, 3, 4]
+    // Expected output: false
+    const hand3 = [5, 5, 5, 5, 4]
+    // Expected output: false
+    const hand4 = [7, 2, 7, 2, 7]
+    // Expected output: true
+    it("should return true for [5, 5, 5, 3, 3]", () => {
+        expect(isFullHouse([5, 5, 5, 3, 3])).toBe("Hand is a full house")
+      })
+    
+      it("should return false for [5, 5, 3, 3, 4]", () => {
+        expect(isFullHouse([5, 5, 3, 3, 4])).toBe("Hand is not a full house")
+      })
+    
+      it("should return false for [5, 5, 5, 5, 4]", () => {
+        expect(isFullHouse([5, 5, 5, 5, 4])).toBe("Hand is not a full house")
+      })
+    
+      it("should return true for [7, 2, 7, 2, 7]", () => {
+        expect(isFullHouse([7, 2, 7, 2, 7])).toBe("Hand is a full house")
+      })
+
+})
+
 
 // b) Create the function that makes the test pass.
+
+//first we need to check if we have five cards
+// if the hand.length is not five return false
+// make an empty object to hold the key value pairs
+// we store a count for each number 
+// iterate through each number in the array using for .. of loop
+// for each count of the number increase the count by 1
+// if the object values include 2, and 3 return is a full house 
+
```

Comment:
Same as JS1, your pseudocode is not fully explaining what is happening in the code. Make sure to elaborate and not leave the pseudocode vague! For example, you say you are making an empty object to hold the key value pairs, but what are the key value pairs? You said you are storing a count for each number, but how?